collated/main/A0135782Y.md

```diff
@@ -0,0 +1,1063 @@
+     * @param endDate The end date of the is task.
+     * @param elapsedYear The years that have elapsed.
+     */
+    private void appendYearlyRecurringTask(ReadOnlyTask task, Calendar startDate,
+            Calendar endDate, int elapsedYear) {
+        // Append a new date to the current task
+        Calendar calendar = Calendar.getInstance();
+        TaskDate editedStartDate = new TaskDate();
+        TaskDate editedEndDate = new TaskDate();
+        if(startDate != null) {
+            calendar.setTime(startDate.getTime());
+            calendar.add(Calendar.YEAR, APPEND_INCREMENT);
+            editedStartDate.setDateInLong(calendar.getTime().getTime());
+            startDate.setTime(editedStartDate.getDate());
+        }else {
+            editedStartDate.setDateInLong((new TaskDate()).getDateInLong());
+        }
+
+        calendar.setTime(endDate.getTime());
+        calendar.add(Calendar.YEAR, APPEND_INCREMENT);
+        editedEndDate.setDateInLong(calendar.getTime().getTime());
+        endDate.setTime(editedEndDate.getDate());
+        
+        TaskComponent newAppendedDate = new TaskComponent((Task) task, editedStartDate, editedEndDate);
+        task.appendRecurringDate(newAppendedDate);
+        repeatingTasks.appendTaskComponent(newAppendedDate);
+    }
+
+    /**
+     * Updates Monthly recurring tasks to the their latest date slot.
+     * 
+     * @param task Recurring task to be considered.
+     * @param startDate The start date of this task if any. Null values represents that start date is not present.
+     * @param endDate The end date of the is task.
+     * @param elapsedYear The months that have elapsed.
+     */
+    private void appendMonthlyRecurringTask(ReadOnlyTask task, Calendar startDate,
+            Calendar endDate, int elapsedMonth) {
+        // Append a new date to the current task
+        // Append a new date to the current task
+        Calendar calendar = Calendar.getInstance();
+        TaskDate editedStartDate = new TaskDate();
+        TaskDate editedEndDate = new TaskDate();
+        if(startDate != null) {
+            calendar.setTime(startDate.getTime());
+            calendar.add(Calendar.MONTH, APPEND_INCREMENT);
+            editedStartDate.setDateInLong(calendar.getTime().getTime());
+            startDate.setTime(editedStartDate.getDate());
+        }else {
+            editedStartDate.setDateInLong((new TaskDate()).getDateInLong());
+        }
+
+        calendar.setTime(endDate.getTime());
+        calendar.add(Calendar.MONTH, APPEND_INCREMENT);
+        editedEndDate.setDateInLong(calendar.getTime().getTime());
+        endDate.setTime(editedEndDate.getDate());
+        
+        TaskComponent newAppendedDate = new TaskComponent((Task) task, editedStartDate, editedEndDate);
+        task.appendRecurringDate(newAppendedDate);
+        repeatingTasks.appendTaskComponent(newAppendedDate);
+    }
```

Comment:
Some methods look very similar. e.g. this method and the previous method. Try to abstract out similarities to reduce duplication.